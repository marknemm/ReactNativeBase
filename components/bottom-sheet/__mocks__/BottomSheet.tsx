import type { BottomSheetProps } from '@components/bottom-sheet/BottomSheet.interfaces';

const MockBottomSheet = jest.fn().mockImplementation((props: BottomSheetProps) => {
  const ActualComponent = jest.requireActual('@components/bottom-sheet/BottomSheet').default;
  return <ActualComponent {...props} />;
});

export default MockBottomSheet;
