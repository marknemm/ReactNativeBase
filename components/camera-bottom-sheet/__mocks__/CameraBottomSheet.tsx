import type { CameraBottomSheetProps } from '@components/camera-bottom-sheet/CameraBottomSheet.interfaces';

const MockCameraBottomSheet = jest.fn().mockImplementation((props: CameraBottomSheetProps) => {
  const ActualComponent = jest.requireActual('@components/camera-bottom-sheet/CameraBottomSheet').default;
  return <ActualComponent {...props} />;
});

export default MockCameraBottomSheet;
