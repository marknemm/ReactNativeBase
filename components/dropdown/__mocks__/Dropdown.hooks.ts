/**
 * Mocks the useDropdownItems hook.
 */
export const useDropdownItems = jest.fn().mockImplementation((data) => data);
