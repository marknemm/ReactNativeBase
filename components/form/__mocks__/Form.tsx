import type { FormProps } from '@components/form/Form.interface';

const FormMock = jest.fn().mockImplementation((props: FormProps) => {
  const ActualComponent = jest.requireActual('@components/form/Form').default;
  return <ActualComponent {...props} />;
});

export default FormMock;
