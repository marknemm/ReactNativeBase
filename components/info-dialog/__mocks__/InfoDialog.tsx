import type { InfoDialogProps } from '@components/info-dialog/InfoDialog.interfaces';

const MockInfoDialog = jest.fn().mockImplementation((props: InfoDialogProps) => {
  const ActualComponent = jest.requireActual('@components/info-dialog/InfoDialog').default;
  return <ActualComponent {...props} />;
});

export default MockInfoDialog;
