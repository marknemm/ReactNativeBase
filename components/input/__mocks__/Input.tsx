import type { InputProps } from '@components/input/Input.interfaces';

const MockInput = jest.fn().mockImplementation((props: InputProps) => {
  const ActualComponent = jest.requireActual('@components/input/Input').default;
  return <ActualComponent {...props} />;
});

export default MockInput;
