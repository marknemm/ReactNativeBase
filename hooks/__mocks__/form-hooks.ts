import { renderHook } from '@testing-library/react-native';
import { useForm, type DefaultValues, type UseFormReturn } from 'react-hook-form';

/**
 * Generate a mock form for testing.
 *
 * @template T The type of the form.
 * @param defaultValues Default values for the form.
 * @returns A mock {@link UseFormReturn}.
 */
export function genMockForm<
  T = DefaultMockFormT
>(
  defaultValues: DefaultValues<T> = { field1: 'value1', field2: 'value2' } as any
): UseFormReturn<T> {
  return renderHook(() =>
    useForm({ defaultValues })
  ).result.current;
}

/**
 * Mock for `useSubmitState` hook.
 */
export const useSubmitState = jest.fn().mockImplementation(() => {
  let submitError = '';
  let submitSuccessful = false;
  let submitting = false;

  return {
    handleSubmit: jest.fn().mockImplementation((submitFn) => async () => { await submitFn(); }),
    handleSubmitState: jest.fn().mockImplementation((submitFn) => async () => { await submitFn(); }),
    setSubmitError: jest.fn().mockImplementation((error) => {
      submitError = error;
    }),
    setSubmitSuccessful: jest.fn().mockImplementation((successful) => {
      submitSuccessful = successful;
    }),
    setSubmitting: jest.fn().mockImplementation((submittingState) => {
      submitting = submittingState;
    }),
    submitError,
    submitSuccessful,
    submitting,
  };
});

/**
 * Default type for the mock form.
 */
export type DefaultMockFormT = {

  /**
   * Form field 1.
   */
  field1: string;

  /**
   * Form field 2.
   */
  field2: string;

};
