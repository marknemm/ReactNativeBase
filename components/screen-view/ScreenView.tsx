import Form from '@components/form/Form';
import { ScrollView, View } from 'react-native';
import type { ScreenViewProps } from './ScreenView.interfaces';
import { useStyles } from './ScreenView.styles';

/**
 * A root {@link View} for a screen.
 *
 * @param props The component {@link ScreenViewProps}.
 * @returns The {@link ScreenView} component.
 */
const ScreenView: React.FC<ScreenViewProps> = ({
  children,
  containerStyle,
  form,
  fullScreen,
  noFooter,
  noHeader,
  noScroll,
  style,
  ...screenProps
}) => {
  const styles = useStyles({ containerStyle, fullScreen, noFooter, noHeader, style });

  const inner = form
    ? (
      <Form
        form={form}
        style={styles.inner}
        testID="rnb-screen-form-view"
      >
        { children }
      </Form>
    )
    : (
      <View
        {...screenProps}
        style={styles.inner}
        testID="rnb-screen-view"
      >
        { children }
      </View>
    );

  return noScroll
    ? (
      <View
        style={styles.container}
        testID="rnb-screen-container-view"
      >
        { inner }
      </View>
    )
    : (
      <ScrollView
        style={styles.container}
        testID="rnb-screen-scroll-container-view"
      >
        { inner }
      </ScrollView>
    );
};

export type * from './ScreenView.interfaces';
export default ScreenView;
