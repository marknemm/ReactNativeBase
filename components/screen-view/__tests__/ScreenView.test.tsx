import ScreenView from '@components/screen-view/ScreenView';
import { genMockForm } from '@hooks/__mocks__/form-hooks';
import { Text } from '@rneui/themed';
import AppProvider from '@test/contexts/app/AppProvider';
import { render, screen } from '@testing-library/react-native';

describe('<ScreenView />', () => {
  describe('display props', () => {
    it('renders standard scroll view', () => {
      render(
        <ScreenView>
          <Text>Test</Text>
        </ScreenView>,
        { wrapper: AppProvider }
      );

      const scrollView = screen.getByTestId('rnb-screen-scroll-container-view');
      expect(scrollView).toBeVisible();
    });

    it('renders children', () => {
      render(
        <ScreenView>
          <Text>Test</Text>
        </ScreenView>,
        { wrapper: AppProvider }
      );

      const scrollView = screen.getByTestId('rnb-screen-scroll-container-view');
      expect(scrollView).toBeVisible();

      const text = screen.getByText('Test');
      expect(text).toBeVisible();
    });

    it('renders form', () => {
      render(
        <ScreenView form={genMockForm()} />,
        { wrapper: AppProvider }
      );

      const formView = screen.getByTestId('rnb-screen-form-view');
      expect(formView).toBeVisible();
    });

    it('renders noScroll', () => {
      render(
        <ScreenView noScroll>
          <Text>Test</Text>
        </ScreenView>,
        { wrapper: AppProvider }
      );

      const screenView = screen.getByTestId('rnb-screen-view');
      expect(screenView).toBeVisible();
    });
  });

  describe('styles', () => {
    it('applies custom styles to scroll container view', () => {
      render(
        <ScreenView containerStyle={{ height: '80%' }}>
          <Text>Test</Text>
        </ScreenView>,
        { wrapper: AppProvider }
      );

      const scrollView = screen.getByTestId('rnb-screen-scroll-container-view');
      expect(scrollView).toHaveStyle({
        height: '80%',
        width: '100%',
      });
    });

    it('applies custom styles to noScroll container view', () => {
      render(
        <ScreenView
          containerStyle={{ height: '80%' }}
          noScroll
        >
          <Text>Test</Text>
        </ScreenView>,
        { wrapper: AppProvider }
      );

      const scrollView = screen.getByTestId('rnb-screen-container-view');
      expect(scrollView).toHaveStyle({
        height: '80%',
        width: '100%',
      });
    });

    it('applies custom styles to screen view', () => {
      render(
        <ScreenView style={{ maxWidth: 800 }}>
          <Text>Test</Text>
        </ScreenView>,
        { wrapper: AppProvider }
      );

      const innerView = screen.getByTestId('rnb-screen-view');
      expect(innerView).toHaveStyle({
        maxWidth: 800,
        width: '100%',
      });
    });

    it('applies custom styles to form view', () => {
      render(
        <ScreenView
          form={genMockForm()}
          style={{ maxWidth: 800 }}
        >
          <Text>Test</Text>
        </ScreenView>,
        { wrapper: AppProvider }
      );

      const innerView = screen.getByTestId('rnb-screen-form-view');
      expect(innerView).toHaveStyle({
        maxWidth: 800,
        width: '100%',
      });
    });
  });

  describe('snapshots', () => {
    it('renders correctly', () => {
      render(
        <ScreenView />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });

    it('renders correctly with children, form, and no scroll', () => {
      render(
        <ScreenView
          form={genMockForm()}
          noScroll
        >
          <Text>Test</Text>
        </ScreenView>,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });
  });
});
