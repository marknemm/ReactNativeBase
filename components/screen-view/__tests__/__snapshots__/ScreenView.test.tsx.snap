// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScreenView /> snapshots renders correctly 1`] = `
<RCTScrollView
  style={
    {
      "height": "100%",
      "width": "100%",
    }
  }
  testID="rnb-screen-scroll-container-view"
>
  <View>
    <View
      style={
        {
          "flex": 1,
          "marginHorizontal": "auto",
          "maxWidth": 1200,
          "paddingBottom": 37.714285714285715,
          "paddingLeft": 18.857142857142858,
          "paddingRight": 18.857142857142858,
          "paddingTop": 37.714285714285715,
          "width": "100%",
        }
      }
      testID="rnb-screen-view"
    />
  </View>
</RCTScrollView>
`;

exports[`<ScreenView /> snapshots renders correctly with children, form, and no scroll 1`] = `
<View
  style={
    {
      "height": "100%",
      "width": "100%",
    }
  }
  testID="rnb-screen-container-view"
>
  <View
    style={
      {
        "flex": 1,
        "marginHorizontal": "auto",
        "maxWidth": 1200,
        "paddingBottom": 37.714285714285715,
        "paddingLeft": 18.857142857142858,
        "paddingRight": 18.857142857142858,
        "paddingTop": 37.714285714285715,
        "width": "100%",
      }
    }
    testID="rnb-screen-form-view"
  >
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#242424",
          "fontFamily": undefined,
          "fontSize": 23.57142857142857,
          "fontStyle": "normal",
          "fontWeight": "normal",
          "marginBottom": 0,
          "textAlign": "left",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
        }
      }
    >
      Test
    </Text>
  </View>
</View>
`;
